lib/json-schema/util/uri.rb

```diff
@@ -1,7 +1,12 @@
+require 'singleton'
+
 module JSON
   module Util
-    module URI
-      def self.normalized_uri(uri)
+    class URI
+      include Singleton
+      attr_accessor :normalize_cache, :parse_cache
```

Comment:
And can't you use the class directly instead of a singleton or switch back to the module and use private cache constants?